front/components/sections/Calculator/helper.ts

```diff
@@ -1,44 +1,62 @@
+import { strapiAppointmentQuery } from 'lib/strapi/appointments/queryType';
 import { formData } from '.';
-import { strapiTherapistQuery } from 'lib/strapi/therapists/queryType';
+import { strapiTherapistsQuery } from 'lib/strapi/therapists/queryType';
 
-export function calculation({ data, therapistList, cost, discount }: props) {
+export const therapyCostCalculation = ({
+    data,
+    therapistList,
+    appointmentList,
+    cost,
+    discount,
+}: props) => {
     discount = 0;
     cost = 0;
+    let numberOfDataSession = Number(data?.session);
     if (data?.workshop) {
-        data.session = Number(data.session) - 1;
+        numberOfDataSession = numberOfDataSession - 1;
     }
-    for (let i = 0; i < Number(therapistList?.length); i++) {
-        if (therapistList?.[i].attributes.first_name === data?.therapist) {
-            cost = Number(therapistList?.[i].attributes.session_cost) * Number(data?.session);
-            if (data?.workshop) {
-                if (Number(data?.session) + 1 >= 24) {
-                    discount = cost * 0.1;
-                    cost = cost * 0.9;
-                }
-            } else {
-                if (Number(data?.session) >= 24) {
-                    discount = cost * 0.1;
-                    cost = cost * 0.9;
-                }
+    therapistList?.find((therapist) => {
+        if (therapist?.id !== Number(data?.therapist)) return;
+
+        cost = Number(therapist.attributes.session_cost) * numberOfDataSession;
+
+        if (data?.workshop) {
+            if (numberOfDataSession + 1 >= 24) {
+                discount = cost * 0.1;
+                cost = cost * 0.9;
+            }
+            discount = discount + Number(therapist.attributes.session_cost);
+        } else {
```

Comment:
```suggestion
        } else if (numberOfDataSession >= 24) {
```
you can omit one nesting